src/snowflake/snowpark/modin/plugin/_internal/indexing_utils.py

```diff
@@ -2215,6 +2218,36 @@ def set_frame_2d_labels(
     index_is_frame = isinstance(index, InternalFrame)
     item_is_frame = isinstance(item, InternalFrame)
     item_is_scalar = is_scalar(item)
+    original_index = index
+    # If `item` is from a Series (rather than a Dataframe), flip the series item values to apply them
```

Comment:
Do we need to update the examples in the docstring?

Reply:
No, we don't need to. I added an example though

Reply:
Sounds good. I'm not sure I can find it though.